graf2d/asimage/src/TASPaletteEditor.cxx

```diff
@@ -467,42 +463,41 @@ void TASPaletteEditor::Open()
    if (!fi.fFilename)
       return;
 
-   TImagePalette *newPalette;
+   TString fn = fi.fFilename;
+
+   TImagePalette *newPalette = nullptr;
 
-   if (strcmp(".pal.txt", fi.fFilename + strlen(fi.fFilename) - 8) == 0) {
-      FILE *fl = fopen(fi.fFilename, "r");
-      if (!fl) return;
-      UInt_t numPoints;
-      // coverity [Calling risky function : FALSE]
-      if (fscanf(fl, "%u\n", &numPoints)) {;}
+   if (fn.EndsWith(".pal.txt")) {
+      FILE *fl = fopen(fn.Data(), "r");
+      if (!fl)
+         return;
+      UInt_t numPoints = 0;
+      if (fscanf(fl, "%u\n", &numPoints) != 1)
+         Error("Open", "Failure reading num points from %s", fn.Data());
       newPalette = new TImagePalette(numPoints);
-      for (Int_t pt = 0; pt < Int_t(numPoints); pt++)
-         // coverity [Calling risky function : FALSE]
-         if (fscanf(fl, "%lf %hx %hx %hx %hx\n",
-                newPalette->fPoints + pt,
-                newPalette->fColorRed + pt,
-                newPalette->fColorGreen + pt,
-                newPalette->fColorBlue + pt,
-                    newPalette->fColorAlpha + pt )) {;}
-             fclose(fl);
+      for (UInt_t pt = 0; pt < numPoints; pt++) {
+         if (fscanf(fl, "%lf %hx %hx %hx %hx\n", newPalette->fPoints + pt, newPalette->fColorRed + pt,
+                    newPalette->fColorGreen + pt, newPalette->fColorBlue + pt, newPalette->fColorAlpha + pt) != 5)
+            Error("Open", "Failure reading point %u from %s", pt, fn.Data());
+      }
+      fclose(fl);
    } else {
       // read from a ROOT file
-      char fn[512];
-      if (strcmp(".pal.root", fi.fFilename + strlen(fi.fFilename) - 9) != 0)
-         snprintf(fn,512, "%s%s", fi.fFilename, ".pal.root");
-      else
-         strlcpy(fn, fi.fFilename,512);
+      if (!fn.EndsWith(".pal.root"))
+         fn.Append(".pal.root");
+
       TDirectory *dirsav = gDirectory;
 
-      TFile *fsave = new TFile(fn, "READ");
+      TFile *fsave = new TFile(fn.Data(), "READ");
       if (!fsave->IsOpen()) {
          delete fsave;
          return;
       }
 
-      newPalette = (TImagePalette*)fsave->Get("TImagePalette");
+      newPalette = (TImagePalette *)fsave->Get("TImagePalette");
```

Comment:
Is it necessary here to allocate on the heap here? Can't we just create a TFile on the stack?

Reply:
One should use `TFile::Open` here. I will adjust commit.